rpc/websockets.go

```diff
@@ -712,8 +712,12 @@ func (api *pubSubAPI) subscribePendingTransactions(wsConn *wsConn) (rpc.ID, erro
 							api.logger.Debug("error writing header, will drop peer", "error", err.Error())
 
 							try(func() {
+								api.filtersMu.RUnlock()
```

Comment:
Please add these comments:
```go
// Release the initial read lock in .RUnlock() before
// invoking .Lock() to avoid the deadlock in
// https://github.com/tharsis/ethermint/issues/821#issuecomment-1033959984
// and as documented at https://pkg.go.dev/sync#RWMutex
```